src/services/rag/components/parsers/markdown.py

```
@@ -40,8 +40,12 @@ async def process(self, file_path: Union[str, Path], **kwargs) -> Document:
 
         self.logger.info(f"Parsing Markdown: {file_path.name}")
 
-        with open(file_path, "r", encoding="utf-8") as f:
-            content = f.read()
+        try:
+            with open(file_path, "r", encoding="utf-8") as f:
+                content = f.read()
+        except OSError as e:
+            self.logger.error(f"Failed to read file {file_path}: {e}")
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)